### PR TITLE
HHH-11797 Add support for @MapKeyEnumerated mappings

### PR DESCRIPTION
**Pull request by Naros**
https://hibernate.atlassian.net/browse/HHH-11797

### CODE REVIEW

File: hibernate-envers/src/main/java/org/hibernate/envers/internal/entities/mapper/relation/MiddleMapKeyEnumeratedComponentMapper.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+	public MiddleMapKeyEnumeratedComponentMapper(
+			AuditEntitiesConfiguration verEntCfg,
+			IdMapper referencedIdMapper,
+			String propertyPrefix,
+			EnumType enumType) {
+		this.verEntCfg = verEntCfg;
+		this.idMapper = referencedIdMapper;
+		this.propertyName = propertyPrefix + "_KEY";
+		this.enumType = enumType;
+	}
```
Comment (dreab8) [inline hibernate-envers/src/main/java/org/hibernate/envers/internal/entities/mapper/relation/MiddleMapKeyEnumeratedComponentMapper.java]: it seems only `propertyPrefix` is needed, may be we can remove the others

File: hibernate-envers/src/main/java/org/hibernate/envers/configuration/internal/metadata/CollectionMetadataGenerator.java [hunk 2]
```diff
@@ -215,7 +217,8 @@ private void addOneToManyAttached(boolean fakeOneToManyBidirectional) {
 		// in a join table, so the prefix value is arbitrary).
 		final MiddleIdData referencedIdData = createMiddleIdData(
 				referencedIdMapping,
-				null, referencedEntityName
+				mappedBy + "_",
```
Comment (dreab8) [inline hibernate-envers/src/main/java/org/hibernate/envers/configuration/internal/metadata/CollectionMetadataGenerator.java]: not sure why this is needed, tried to change it back to null and all the tests passed

File: hibernate-envers/src/main/java/org/hibernate/envers/internal/entities/mapper/relation/MiddleIdData.java [hunk 3]
```diff
@@ -26,7 +26,7 @@ public MiddleIdData(
 			AuditEntitiesConfiguration verEntCfg, IdMappingData mappingData, String prefix,
 			String entityName, boolean audited) {
 		this.originalMapper = mappingData.getIdMapper();
-		this.prefixedMapper = mappingData.getIdMapper().prefixMappedProperties( prefix );
+		this.prefixedMapper = prefix == null ? null : mappingData.getIdMapper().prefixMappedProperties( prefix );
```
Comment (dreab8) [inline hibernate-envers/src/main/java/org/hibernate/envers/internal/entities/mapper/relation/MiddleIdData.java]: as for the previous comment I'm not sure why this is needed, I tried to change it back to `this.prefixedMapper = mappingData.getIdMapper().prefixMappedProperties( prefix );` and all the tests passed

**Comment by Naros** [comment]
Hi @dreab8, I'll double check your comments.  If I can't see any reason why they should be there I'll change the code and push to the PR so you can make the release this weekend.

**Comment by Naros** [comment]
@dreab8 Great catches on all fronts, fixed the changes as suggested, rebased and squashed.  It should be all set for merging if CI passes.

**Comment by dreab8** [comment]
thanks @Naros

**Pull request closed without merging**